presto-hive/src/main/java/com/facebook/presto/hive/HiveSplitManager.java

```diff
@@ -356,12 +356,18 @@ public CounterStat getHighMemorySplitSource()
                 if ((tableColumns == null) || (partitionColumns == null)) {
                     throw new PrestoException(HIVE_INVALID_METADATA, format("Table '%s' or partition '%s' has null columns", tableName, partName));
                 }
-                ImmutableMap.Builder<Integer, HiveTypeName> columnCoercions = ImmutableMap.builder();
-                for (int i = 0; i < min(partitionColumns.size(), tableColumns.size()); i++) {
+                ImmutableMap.Builder<Integer, Column> columnCoercions = ImmutableMap.builder();
```

Comment:
maybe rename all the places that say columnCoercions to something like `columnDiffs` to make clear that these are no longer just type coercions, but contain the diff of anything different between the partition columns and the table columns

Reply:
Renamed to `partitionSchemaDifference`